contracts/src/components/hyperstructure.cairo

```diff
@@ -0,0 +1,20 @@
+#[derive(Component, Copy, Drop, Serde, SerdeLen)]
+struct HyperStructure {
+    #[key]
+    entity_id: u128,
+    created_at: u64,
+    completed_at: u64,
+    resource_count: usize
```

Comment:
let's add a 'hyperstructure_id' here too to define the type

Reply:
@credence0x as loaf noted, in the future we migth want to have different types of hyperstructures. That can give different bonuses if completed. 
So adding a hyperstructure_type field here would help us differentiate between these hyperstructures. 

Let's say the ones we create for each order are going to be the first hyperstrucutres. And let's give them hyperstructure_type 1.